sdk/hatchery/register.go

```diff
@@ -125,11 +125,33 @@ func Create(h Interface) {
 				// count + 1 here, and remove -1 if worker is not started
 				// this avoid to spawn to many workers compare
 				atomic.AddInt64(&workersStarted, 1)
-				if isRun := receiveJob(h, true, nil, job.ID, job.QueuedSeconds, job.BookedBy, job.Job.Action.Requirements, models, &nRoutines, spawnIDs, hostname); isRun {
+				if isRun, errRec := receiveJob(h, true, nil, job.ID, job.QueuedSeconds, job.SpawnAttempts, job.BookedBy, job.Job.Action.Requirements, models, &nRoutines, spawnIDs, hostname); isRun {
 					atomic.AddInt64(&workersStarted, 1)
 					spawnIDs.SetDefault(string(job.ID), job.ID)
-				} else {
+				} else if errRec == nil {
 					atomic.AddInt64(&workersStarted, -1)
+
+					if hCount, err := h.Client().HatcheryCount(job.WorkflowNodeRunID); err == nil {
```

Comment:
I think that this code (store sdk.MsgSpawnInfoHatcheryCannotStartJob.ID ) should be done by API only:

```
if errRec == nil
 --> call QueueJobIncAttemps
```

and QueueJobIncAttemps should write MsgSpawnInfoHatcheryCannotStartJob if nbAttemps > HatcheryCount 

Each hatchery have to remember on which job they store attemps, to avoid inc many time same job, no?